docs/changes/2234.feature.rst

```diff
@@ -0,0 +1,5 @@
+``QualityQuery`` is now a ``TelescopeComponent``.
+
+You can now give different quality queries per telescope type or id.
+Change is backwards compatible, older configurations still work and
+set the query-expression to the global value of the TelescopeComponent.
```

Comment:
Since `QualityQuery` now require the subarray, it's not fully API compatible. Only with respect to that existing configurations will still work the same for tools that were updated.

Which makes me wonder if we instead of making the `QualityQuery` a telescope component we should instead add a `TelescopeQualityQuery`... Do have have a use-case for a `QualityQuery` that does not need a subarray?

Reply:
> Since QualityQuery now require the subarray, it's not fully API compatible. Only with respect to that existing configurations will still work the same for tools that were updated.

Right, will change that.

> Do have have a use-case for a QualityQuery that does not need a subarray?

We have `ctapipe.reco.reconstructor.StereoQualityQuery`, which does not necessarily need a subarray I think

Reply:
Setting `subarray=None` (which I now implemented as default) can be used as a ~`Telescope`~`Component`.  Giving a `key` on calling fails with an okayish error:

```
ValueError: TelescopeParameterLookup: No subarray attached, call `attach_subarray` first before trying to access a value by tel_id
```